OnboardPietroPieri/components/user-detail.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useState } from "react";
+import { View, Text, StyleSheet, ActivityIndicator } from "react-native";
+import { ParamListBase, RouteProp } from "@react-navigation/native";
+import { NativeStackScreenProps } from "@react-navigation/native-stack";
+import { useQuery } from "@apollo/client";
+import { USER } from "../apollo/query";
+import { UserOutput } from "../interfaces/query";
+import { User } from "../interfaces/mutation";
+import { BackButton } from "./back-button";
+
+type UserDetailRouteProp = RouteProp<ParamListBase, "UserDetail">;
+
+type Props = NativeStackScreenProps<ParamListBase, "UserDetail">;
+
+export const UserDetail: React.FC<Props> = ({
+  route,
+}: {
+  route: UserDetailRouteProp;
+}) => {
+  const { id } = (route.params as { id: string }) || { id: undefined };
```

Comment:
Nit: acredito que vá produzir o mesmo resultado
```suggestion
  const { id } = (route.params || {}) as { id?: string };
```